js/lib/DateFmt.js

```diff
@@ -437,60 +451,75 @@ var DateFmt = function(options) {
             // get the default calendar name from the locale, and if the locale doesn't define
             // one, use the hard-coded gregorian as the last resort
             this.calName = this.calName || this.locinfo.getCalendar() || "gregorian";
-            if (ilib.isDynCode()) {
-                // If we are running in the dynamic code loading assembly of ilib, the following
-                // will attempt to dynamically load the calendar date class for this calendar. If
-                // it doesn't work, this just goes on and it will use Gregorian instead.
-                DateFactory._init(this.calName);
+            
+            if(sync && typeof(Intl) !== 'undefined' && Intl.DateTimeFormat.supportedLocalesOf(this.locale.getSpec()).length > 0){
+                if (this.locinfo.getDigitsStyle() == "western" && (!options.template) && this.calName == "gregorian")    {
+                    if(this.type == "date" && ((this.dateComponents == "dmy" && this.length != "f") ||
```

Comment:
All three lines can be one long "if" condition.

Also, we don't need to check for sync. We can create the Intl date formatter synchronously and then call the callback as if it were asynchronous.

`this.type == "date"` -> we can also format times with the Intl formatter, can't we?

Reply:
Yes, it is a middle of progress that haven't written completely. I just made a draft PR for the preparing discussion our meeting this week (Unfortunately we missed it). I'm not sure if you had chance to read mail that I send to you last Tuesday, I wrote some issues regarding the implementation these way.  Let's discuss these in next meeting and then will continue to work on.